dds_cli/data_putter.py

```diff
@@ -169,6 +170,75 @@ def put(
                         for x in [y.id for y in progress.tasks if y.fields.get("step") != "put"]
                     ]
 
+        # Make a single database update for files that have failed
+        # Json file for failed files should only be created if there has been an error
+        if putter.failed_delivery_log.is_file():
+            LOG.warning(
+                "Some uploaded files could not be added to the database %s.\n",
+                str(putter.failed_delivery_log),
+            )
+            LOG.warning("Attempting to add them to the database.")
+
+            with putter.failed_delivery_log.open(mode="r", encoding="utf-8") as json_f:
+                failed = json.load(json_f)
+
+            # remove from log any files that  failed for other reasons
+            for file, values in failed.items():
+                if values.get("status", {}).get("failed_op") != "add_file_db":
+                    failed.pop(file)
+
+            # send failed to API endpoint
+            params = {"project": putter.project}
+            try:
+                response, _ = dds_cli.utils.perform_request(
+                    DDSEndpoint.FILE_UPDATE_FAILED,
+                    method="put",
+                    headers=putter.token,
+                    params=params,
+                    json=failed,
+                    error_message="Failed to add missing files",
+                )
+            except (
+                dds_cli.exceptions.ApiRequestError,
+                dds_cli.exceptions.ApiResponseError,
+                dds_cli.exceptions.DDSCLIException,
+            ) as err:
+                LOG.warning(str(err))
+
+            message = response.get("message")
+            if not message:
+                raise dds_cli.exceptions.ApiResponseError(
+                    message="Failed to add missing files to database."
+                )
+            # check if files are uploaded in db
+            files = list(failed.keys())
+            files_in_db, _ = dds_cli.utils.perform_request(
```

Comment:
Probably don't need to do this check anymore since we're returning if there's an error